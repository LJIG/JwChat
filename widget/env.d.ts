/*
 * @Author            : Bian <389701057@qq.com>
 * @Date              : 2025-06-24 22:44:04
 * @LastEditors       : LJIG <389701057@qq.com>
 * @LastEditTime      : 2025-06-24 22:44:05
 * @FilePath          : /packages/env.d.ts
 * @Description       : 
 * Copyright (c) 2025 by Bian <389701057@qq.com>, All Rights Reserved. 
 */

declare module "*.vue" {
  import { DefineComponent } from "vue";
  const component: DefineComponent<{}, {}, any>;
  export default component;
}

declare module "*.md" {
  import { ComponentOptions } from "vue";
  const Component: ComponentOptions;
  export default Component;
}

declare module "wechat-emoji-parser";

declare module "@/utils/scroll";

declare module "*.scss" {
  const classes: { readonly [key: string]: string };
  export default classes;
}

declare module "*.png";
declare module "@/utils/imgUrl";
