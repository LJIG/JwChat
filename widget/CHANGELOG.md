## 0.0.2 (2022-04-22)


### Bug Fixes

* fix template ([21a31e4](https://github.com/jrainlau/MY-Kit/commit/21a31e41647fadef0e458539476805c28bd59e69))


### Features

* chat组件 ([e4abb9a](https://github.com/jrainlau/MY-Kit/commit/e4abb9a5a12b785b4ff23e843e37263e32ffca17))


### BREAKING CHANGES

* 突破从无到有



