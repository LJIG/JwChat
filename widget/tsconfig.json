{
  "compilerOptions": {
    "noImplicitAny": false, // 是否在表达式和声明上有隐含的any类型时报错
    "target": "esnext",
    "useDefineForClassFields": true,
    "module": "esnext",
    "moduleResolution": "node",
    "strict": true,
    "jsx": "preserve",
    "sourceMap": true,
    "resolveJsonModule": true,
    "esModuleInterop": true,
    "allowJs": true,
    "lib": ["esnext", "dom"],
    "skipLibCheck": true,
    "paths": {
      "packages/*": ["./packages/*"]
    }
  },
  "include": [
    "*.ts",
    "packages/**/*.ts",
    "packages/**/*.d.ts",
    "packages/**/*.tsx",
    "packages/**/*.vue"
  ]
}
