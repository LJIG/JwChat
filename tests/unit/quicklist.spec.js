import '../util.js'

import { mount  } from '@vue/test-utils'
import widget from '@/components/QuickTalk/index.vue'

describe('QuickList 组件', () => {

  const config = {
    nav: ['快捷回复', '超级回复'],
    showAdd: true,
    maxlength: 200,
    showHeader: true,
    showDeleteBtn: true,
  }

  const Talelist = [
    '快捷回复1',
    '快捷回复2',
    '快捷回复3',
    '快捷回复4',
    '快捷回复5',
    '快捷回复6',
  ]

  const wrapper = mount(widget, { propsData: { config, Talelist } })
  // 使用 jest 我们可以创建HTML输出的截图文件
  expect(wrapper.html()).toMatchSnapshot()


  it('默认菜单选中', (done) => {

    const defaultMenu = wrapper.find('.el-menu-demo .is-active')
    expect(defaultMenu.text()).toBe(config.nav[0])
    done()
    
  })

  it('切换选中菜单',async ()=>{

    const menus = wrapper.findAll('.el-menu-demo li')
    
    // 模拟按钮的点击操作
    menus.at(1).trigger('click')
    await sleep()
    expect(wrapper.vm.activeIndex).toBe('2')
    
    menus.filter(w => w.text() == config.nav[0]).trigger('click')
    expect(wrapper.vm.activeIndex).toBe('1')

  })

})

function sleep(time=1000) {
  return new Promise(resolve=>{
    setTimeout(() => {
      resolve()
    }, time);
  })
}