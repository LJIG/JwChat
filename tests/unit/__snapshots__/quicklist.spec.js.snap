// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div class="wrapper">
  <transition-stub mode="out-in" class="el-menu-demo">
    <ul role="menubar" class="el-menu--horizontal el-menu" tabindex="0">
      <li role="menuitem" tabindex="-1" class="el-menu-item is-active">快捷回复</li>
      <li role="menuitem" tabindex="-1" class="el-menu-item" style="border-bottom-color: transparent;">超级回复</li> <i title="新增" class="addBtn el-icon-circle-plus-outline" style="visibility: visible;"></i>
    </ul>
  </transition-stub>
  <!---->
  <ul>
    <li>
      <div class="el-row">
        <div class="el-col el-col-19">
          <p>快捷回复1</p>
        </div>
        <div class="el-col el-col-5"><i title="发送" class="el-icon-circle-check"></i> <i title="删除" class="el-icon-circle-close"></i></div>
      </div>
    </li>
    <li>
      <div class="el-row">
        <div class="el-col el-col-19">
          <p>快捷回复2</p>
        </div>
        <div class="el-col el-col-5"><i title="发送" class="el-icon-circle-check"></i> <i title="删除" class="el-icon-circle-close"></i></div>
      </div>
    </li>
    <li>
      <div class="el-row">
        <div class="el-col el-col-19">
          <p>快捷回复3</p>
        </div>
        <div class="el-col el-col-5"><i title="发送" class="el-icon-circle-check"></i> <i title="删除" class="el-icon-circle-close"></i></div>
      </div>
    </li>
    <li>
      <div class="el-row">
        <div class="el-col el-col-19">
          <p>快捷回复4</p>
        </div>
        <div class="el-col el-col-5"><i title="发送" class="el-icon-circle-check"></i> <i title="删除" class="el-icon-circle-close"></i></div>
      </div>
    </li>
    <li>
      <div class="el-row">
        <div class="el-col el-col-19">
          <p>快捷回复5</p>
        </div>
        <div class="el-col el-col-5"><i title="发送" class="el-icon-circle-check"></i> <i title="删除" class="el-icon-circle-close"></i></div>
      </div>
    </li>
    <li>
      <div class="el-row">
        <div class="el-col el-col-19">
          <p>快捷回复6</p>
        </div>
        <div class="el-col el-col-5"><i title="发送" class="el-icon-circle-check"></i> <i title="删除" class="el-icon-circle-close"></i></div>
      </div>
    </li>
  </ul>
  <transition-stub name="dialog-fade">
    <div class="el-dialog__wrapper" style="display: none;">
      <div role="dialog" aria-modal="true" aria-label="新增快捷回复" class="el-dialog" style="margin-top: 15vh; width: 36%;">
        <div class="el-dialog__header"><span class="el-dialog__title">新增快捷回复</span><button type="button" aria-label="Close" class="el-dialog__headerbtn"><i class="el-dialog__close el-icon el-icon-close"></i></button></div>
        <!---->
        <!---->
      </div>
    </div>
  </transition-stub>
</div>
`;
